File: 4기-코딩테스트-220608/4기 코딩테스트_220608/4기 - 1st 코딩테스트/stock/stock.js
function solution(A)
{
  var result = [];

  for (var i = 0; i < A.length - 1; i++)
  {
    for (var j = i + 1; j < A.length; j++)
    {
      if (A[i] < A[j])
      {
        result.push(A[j] - A[i]);
      }
    }
  }

  return result.length == 0 ? 0 : Math.max.apply(null, result);
}

File: 4기-코딩테스트-220608/4기 코딩테스트_220608/4기 - 1st 코딩테스트/stock/stock.py
def solution(A):
    stock = [0 for _ in range(len(A))]
    stack = [0]

    for i in range(1, len(A)):
        for j in range (0, i):
            temp = A[i] - A[j]
            if temp > 0 and stock[j] < temp:
                stock[j] = temp

    return max(stock)
